src/app/label-data-validation/page.tsx

```diff
@@ -1,29 +1,153 @@
+
   return (
-    <Container className="flex flex-col h-screen max-w-[1920px] overflow-hidden" maxWidth={false}>
-      <Box className="p-4">
-        stepper
+    <Container
+      className="flex flex-col h-screen max-w-[1920px] overflow-hidden"
+      maxWidth={false}
+    >
+      <Box className="p-4 mt-4">
+        <HorizontalNonLinearStepper
+          steps={steps}
+          activeStep={activeStep}
+          setActiveStep={setActiveStep}
+          stepStatuses={stepStatuses}
+        />
       </Box>
 
       <Box className="flex flex-1 overflow-hidden flex-col md:flex-row">
         <Box className="flex-1 p-4 flex justify-center">
           {/* Carousel Placeholder: To be removed, just for testing */}
-          <Box className="w-full p-4 text-center font-bold bg-gray-300">
+          <Box className="w-full h-3/4 p-4 text-center font-bold bg-gray-300">
             Carousel Placeholder
           </Box>
         </Box>
 
         <Box className="flex flex-1 p-4 justify-center overflow-y-auto">
           {/* Form Placeholder: To be removed, just for testing */}
-          <Box className="w-full h-[130vh] p-4 text-center font-bold bg-gray-400">
-            Form Placeholder
+          <Box className="w-full h-3/4 p-4 text-center font-bold bg-gray-400">
+            <StepControls
+              steps={steps}
+              activeStep={activeStep}
+              setActiveStep={setActiveStep}
+              stepStatuses={stepStatuses}
+              setStepStatuses={setStepStatuses}
+            />
           </Box>
         </Box>
       </Box>
     </Container>
   );
 }
 
+// To be removed, just for testing
+const StepControls: React.FC<CustomStepperProps> = ({
+  steps,
+  activeStep,
+  setActiveStep,
+  stepStatuses,
+  setStepStatuses,
+}) => {
+  const stepsTotal = steps.length;
+  const allStepsCompleted = Object.values(stepStatuses).every(
+    (status) => status === StepStatus.Completed,
+  );
+
+  const handleNext = () => {
+    setActiveStep((prev) => Math.min(prev + 1, stepsTotal - 1));
+  };
+
+  const handleBack = () => {
+    setActiveStep((prev) => Math.max(prev - 1, 0));
+  };
+
+  const handleComplete = () => {
+    setStepStatuses?.((prev) => ({
+      ...prev,
+      [activeStep]: StepStatus.Completed,
+    }));
+  };
+
+  const handleIncomplete = () => {
+    setStepStatuses?.((prev) => ({
+      ...prev,
+      [activeStep]: StepStatus.Incomplete,
+    }));
+  };
+
+  const handleError = () => {
+    setStepStatuses?.((prev) => ({ ...prev, [activeStep]: StepStatus.Error }));
+  };
+
+  const handleReset = () => {
+    setActiveStep(0);
+    setStepStatuses?.({});
+  };
+
+  return (
+    <>
+      <Typography sx={{ mt: 2, mb: 1, py: 1 }}>
+        {allStepsCompleted
+          ? "All steps completed - you're finished"
+          : `Step ${activeStep + 1}`}
+      </Typography>
+
+      <Box sx={{ display: "flex", justifyContent: "center", gap: 2, pt: 2 }}>
+        <Button
```

Comment:
Tailwind classes?

Reply:
Done. Will update other components not part of this PR as well, progressively.